src/main/java/bio/terra/service/dataset/DatasetService.java

```diff
@@ -353,6 +355,91 @@ public String addDatasetAssetSpecifications(
         .submit();
   }
 
+  public void validateDatasetAssetSpecification(Dataset dataset, AssetModel assetModel) {
```

Comment:
Can this be made `static`? It doesn't look like it depends on any fields in `DatasetService`. Actually, if it can be `static`, it might make the most sense as a non-static method on `Dataset`.